src/main/java/Duke.java

```diff
@@ -1,10 +1,274 @@
+        public void execute(ArrayList<Task> list, String description) throws DukeException {
+            String[] arr = description.split(" ", 2);
+            if (arr.length <= 1) {
+                throw new DukeException("Please use the format: deadline <task> /by <date/time>");
+            }
+
+            String[] dlarr = arr[1].split("/", 2);
+            if (dlarr.length <= 1) {
+                throw new DukeException("Please use the format: deadline <task> e/by <deadline>.");
+            }
+            String name = dlarr[0];
+            String by = dlarr[1];
+            Task task = new Deadline(name, by);
+            list.add(task);
+            int n = list.size();
+            System.out.println("added: " + task);
+            System.out.println("You now have " + n + " tasks.");
+        }
+
+    },
+    TODO("todo") {
+        @Override
+        public void execute() {
+            return;
+        }
+
+        @Override
+        public void execute(ArrayList<Task> list, String description) throws DukeException {
+            String[] arr = description.split(" ", 2);
+            if (arr.length <= 1) {
+                throw new DukeException("Please use the format: todo <task>");
+            }
+            Task task = new Todo(arr[1]);
+            list.add(task);
+            int n = list.size();
+            System.out.println("added: " + task);
+            System.out.println("You now have " + n + " tasks.");
+
+        }
+
+    },
+    UNKNOWN("unknown") {
+
+        @Override
+        public void execute() throws DukeException {
+            throw new DukeException("I don't know what that means");
+        }
+        @Override
+        public void execute(ArrayList<Task> list, String description) {
+            return;
+        }
+
+    };
+
+    private final String command;
+    Commands(String command) {
+        this.command = command;
+    }
+    public String getCommand() {
+        return this.command;
+    }
+    public abstract void execute() throws DukeException;
+    public abstract void execute(ArrayList<Task> list, String description) throws DukeException;
+}
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    private static ArrayList<Task> list = new ArrayList<>();
+
+    public static void main(String[] args) throws DukeException {
+
+        Commands currentCommand = Commands.HELLO;
+        currentCommand.execute();
+
+        Scanner scanner = new Scanner(System.in);
+
+        while (scanner.hasNextLine()) {
+            String str = scanner.nextLine();
+            String[] arr = str.split(" ", 2);
+            String keyword = arr[0].toLowerCase();
+
+            switch (keyword) {
```

Comment:
In line with conventions, might it make more correct to reduce the indentation on the `case`?